snark-verifier-sdk/examples/halo2_lib.rs

```diff
@@ -0,0 +1,104 @@
+use ark_std::{end_timer, start_timer};
+use halo2_base::gates::builder::{CircuitBuilderStage, BASE_CONFIG_PARAMS};
+use halo2_base::utils::fs::gen_srs;
+
+use snark_verifier_sdk::halo2::read_snark;
+use snark_verifier_sdk::SHPLONK;
+use snark_verifier_sdk::{
+    gen_pk,
+    halo2::{aggregation::AggregationCircuit, gen_snark_shplonk},
+    Snark,
+};
+use std::fs::File;
+use std::io::{BufReader, BufWriter};
+use std::path::Path;
+
+fn read_snark_from_file(file_name: &str) -> Snark {
+    let snark_path = Path::new(file_name);
+    let snark = read_snark(snark_path)
+        .unwrap_or_else(|e| panic!("Snark not found at {snark_path:?}. {e:?}"));
+    snark
+}
+
+fn gen_agg_break_points(agg_circuit: AggregationCircuit, path: &Path) -> Vec<Vec<usize>> {
+    let file = File::open(path);
+    let break_points = match file {
+        Ok(file) => {
+            let reader = BufReader::new(file);
+            let break_points: Vec<Vec<usize>> = serde_json::from_reader(reader).unwrap();
+            break_points
+        }
+        Err(_) => {
+            let break_points = agg_circuit.break_points();
+            let file = File::create(path).unwrap();
+            let writer = BufWriter::new(file);
+            serde_json::to_writer(writer, &break_points).unwrap();
+            break_points
+        }
+    };
+    break_points
+}
+
+fn main() {
+    let dummy_snark = read_snark_from_file("./examples/halo2_lib_snarks/range.snark");
+
+    let k = 14u32;
+    let params = gen_srs(k);
+    let lookup_bits = k as usize - 1;
+    BASE_CONFIG_PARAMS.with(|config| {
+        config.borrow_mut().lookup_bits = Some(lookup_bits);
+        config.borrow_mut().k = k as usize;
+    });
+    let agg_circuit = AggregationCircuit::new::<SHPLONK>(
+        CircuitBuilderStage::Keygen,
+        None,
+        lookup_bits,
+        &params,
+        vec![dummy_snark.clone()],
+        true,
+    );
+    agg_circuit.config(k, Some(10));
+
+    let start0 = start_timer!(|| "gen vk & pk");
+    let pk = gen_pk(&params, &agg_circuit, Some(Path::new("./examples/agg.pk")));
```

Comment:
agg.pk might be used for other examples, let's name it something more descriptive like univ_lib_agg or something?